Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,793 @@
+                        Write-Warning "AMSI is NOT disabled on $server"
+                    } else {
+                        Write-Host "Removing SettingOverride $($getSO.Identity)"
+                        $getSO | Out-Host
+                        Remove-SettingOverride -Identity $getSO.Identity -Confirm:$false
+                        Write-Host "Enabled on $server" -ForegroundColor Green
+                        Get-ExchangeDiagnosticInfo -Process Microsoft.Exchange.Directory.TopologyService -Component VariantConfiguration -Argument Refresh -Server $server | Out-Null
+                        Write-Host "Refreshed Get-ExchangeDiagnosticInfo on $server"
+                        $needsRefresh++
+                    }
+                }
+            } else {
+                Write-Host $bar
+                Write-Host ""
+                $getSO = Get-SettingOverride -ErrorAction SilentlyContinue | Where-Object {
+                    ( $_.SectionName -eq 'HttpRequestFiltering' ) -and
+                    ( $_.Parameters -eq 'Enabled=False' ) -and
+                    ( $null -eq $_.Server ) }
+                if ( $null -eq $getSO ) {
+                    Write-Host "We did not find Get-SettingOverride that disabled AMSI at Organization level"
+                    Write-Warning "AMSI is NOT disabled on Exchange configuration at organization level"
+                } else {
+                    Write-Host "Removing SettingOverride $($getSO.Identity)"
+                    $getSO | Out-Host
+                    Remove-SettingOverride -Identity $getSO.Identity -Confirm:$false
+                    Write-Host "Enabled AMSI at Organization Level" -ForegroundColor Green
+                    foreach ($server in $filterList) {
+                        Get-ExchangeDiagnosticInfo -Process Microsoft.Exchange.Directory.TopologyService -Component VariantConfiguration -Argument Refresh -Server $server | Out-Null
+                        Write-Host "Refreshed Get-ExchangeDiagnosticInfo on $server"
+                    }
+                    $needsRefresh++
+                }
+            }
+        }
+
+        if ($DisableAMSI) {
+            $getSO = $null
+            if ($filterList) {
+                foreach ($server in $filterList) {
+                    Write-Host $bar
+                    Write-Host ""
+                    $getSO = Get-SettingOverride -ErrorAction SilentlyContinue | Where-Object {
+                        ( $_.SectionName -eq 'HttpRequestFiltering' ) -and
+                        ( $_.Parameters -eq 'Enabled=False' ) -and
+                        ( $_.Server -contains $server ) }
+                    if ( $null -eq $getSO ) {
+                        New-SettingOverride -Name "DisablingAMSIScan-$server" -Component Cafe -Section HttpRequestFiltering -Parameters ("Enabled=False") -Reason "Disabled via CSS-Exchange Script" -Server $server
```

Comment:
Maybe we should use `$PSCmdlet.ShouldProcess` here? Not sure how `New-SettingOverride` behaves if someone uses `WhatIf`.
https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-shouldprocess?view=powershell-7.3

Reply:
I changed it to show:
![image](https://github.com/microsoft/CSS-Exchange/assets/103440830/67fcd8b9-6d10-424e-8c43-7507d407b2a0)